x/mock/app.go

```diff
@@ -206,9 +206,8 @@ func SetGenesis(app *App, accs []authexported.Account) {
 	app.Commit()
 }
 
-// GenTx generates a signed mock transaction.
-func GenTx(msgs []sdk.Msg, accnums []uint64, seq []uint64, priv ...crypto.PrivKey) auth.StdTx {
-	// Make the transaction free
+func GenTxWithChainID(cid string, msgs []sdk.Msg, accnums []uint64, seq []uint64, priv ...crypto.PrivKey) auth.StdTx {
```

Comment:
these structs will be deleted for the ones that are located in `simapp/test_helpers.go`. We may want to dup this func there as well